safety/Register-Patient-Safety.ps1

```diff
@@ -0,0 +1,272 @@
+        name = "$name"
+        grain = "$grain"
+        securableItem = "$securableItem"
+    }
+    return Invoke-Post $url $body $accessToken
+}
+
+function Add-Group($authUrl,$name, $source, $accessToken)
+{
+    $url = "$authUrl/groups"
+    $body = @{
+        id = "$name"
+        groupName = "$name"
+        groupSource = "$source"
+    }
+    return Invoke-Post $url $body $accessToken
+}
+
+function Add-PermissionToRole($authUrl, $roleId, $permission, $accessToken)
+{
+    $url = "$authUrl/roles/$roleId/permissions"
+    $body = @($permission)
+    return Invoke-Post $url $body $accessToken
+}
+
+function Add-RoleToGroup($authUrl, $groupName, $role, $accessToken)
+{
+    $encodedGroupName = [System.Web.HttpUtility]::UrlEncode($groupName)
+    $url = "$authUrl/groups/$encodedGroupName/roles"
+    $body = $role
+    return Invoke-Post $url $body $accessToken
+}
+
+if(!(Test-Path .\Fabric-Install-Utilities.psm1)){
+	Invoke-WebRequest -Uri https://raw.githubusercontent.com/HealthCatalyst/InstallScripts/master/common/Fabric-Install-Utilities.psm1 -OutFile Fabric-Install-Utilities.psm1
+}
+Import-Module -Name .\Fabric-Install-Utilities.psm1 -Force
+
+if(!(Test-IsRunAsAdministrator))
+{
+    Write-Error "You must run this script as an administrator. Halting configuration." -ErrorAction Stop
+}
+
+$installSettings = Get-InstallationSettings "registration"
+$fabricInstallerSecret = $installSettings.fabricInstallerSecret
+$authorizationSerivceURL =  $installSettings.authorizationSerivce
+$identityServerUrl = $installSettings.identityService
+$safetySurveillanceUrl = $installSettings.safetySurveillanceService
+$groupName = $installSettings.groupName
+
+if([string]::IsNullOrEmpty($installSettings.identityService))  
+{
+	$identityServerUrl = Get-IdentityServiceUrl
+} else
+{
+	$identityServerUrl = $installSettings.identityService
+}
+
+if([string]::IsNullOrEmpty($installSettings.authorizationSerivce))  
+{
+	$authorizationSerivceURL = Get-AuthorizationServiceUrl
+} else
+{
+	$authorizationSerivceURL = $installSettings.authorizationSerivce
+}
+
+if([string]::IsNullOrEmpty($installSettings.safetySurveillanceService))  
+{
+	$safetySurveillanceUrl = Get-SafetySurveillanceUrl
+} else
+{
+	$safetySurveillanceUrl = $installSettings.safetySurveillanceService
+}
+
+try{
+	$encryptionCert = Get-Certificate $encryptionCertificateThumbprint
+}catch{
+	Write-Host "Could not get encryption certificate with thumbprint $encryptionCertificateThumbprint. Please verify that the encryptionCertificateThumbprint setting in install.config contains a valid thumbprint for a certificate in the Local Machine Personal store. Halting installation."
+	throw $_.Exception
+}
+
+
+$userEnteredFabricInstallerSecret = Read-Host  "Enter the Fabric Installer Secret or hit enter to accept the default [$fabricInstallerSecret]"
+Write-Host ""
+if(![string]::IsNullOrEmpty($userEnteredFabricInstallerSecret)){   
+     $fabricInstallerSecret = $userEnteredFabricInstallerSecret
+}
+
```

Comment:
Do we want to throw an error if `$fabricInstallerSecret, $identityServerUrl, or $authorizationSerivceURL` is empty after the user input is read?

Reply:
Good call, I'll add in that check.